### ISSUE TITLE
TextDecoration.lineThrough not working

### ISSUE THREAD

**Issue by premtemp1** [daohoangson/flutter_widget_from_html#301]
Hello,
Thanks for the widget.. However, I am not able to get "TextDecoration.lineThrough" to work.
Could you please let me how to do strikeout with your widget.

Thanks

child: HtmlWidget(
                messageModel.replies,
                textStyle: TextStyle(
                  decoration: (messageModel.shouldDelete)
                      ? TextDecoration.lineThrough
                      : TextDecoration.none,
                  fontSize: 15.0,
                  height: 1.4,
                  color: Colors.white,
                ),
              ),

It work when rendering the widget initially. However, if shouldDelete is change to true and you call setstate then it does not re-render correctly

**Comment by daohoangson** [comment]
This is a known issue with `v0.4`. It has been fixed in `v0.5` though (pre-release version available, stable coming soon). The example below runs with latest 0.5 from pub.dev:

<details>

<summary>test.dart</summary>

```
class Test extends StatefulWidget {
  @override
  _TestState createState() => _TestState();
}

class _TestState extends State<Test> {
  bool lineThrough = false;

  @override
  Widget build(BuildContext context) => Scaffold(
        appBar: AppBar(title: Text('Issue 301')),
        body: Padding(
          child: Column(
            children: [
              CheckboxListTile(
                title: Text('lineThough?'),
                value: lineThrough,
                onChanged: (v) => setState(() => lineThrough = v),
              ),
              Expanded(
                child: SingleChildScrollView(
                  child: HtmlWidget(
                    'Hello, <strong>world</strong>',
                    textStyle: TextStyle(
                      decoration: lineThrough
                          ? TextDecoration.lineThrough
                          : TextDecoration.none,
                      fontSize: 15.0,
                      height: 1.4,
                    ),
                  ),
                ),
              ),
            ],
          ),
          padding: const EdgeInsets.all(8.0),
        ),
      );
}

```

</details>

<img src="https://user-images.githubusercontent.com/239336/91628475-3b9d3900-e9ea-11ea-885a-4dc4c64e7723.gif" width="300" />

If you are not comfortable running pre-release package in your app, it's possible to workaround by using a `Key` for your `HtmlWidget`, something like this:

```dart
                  HtmlWidget(
                    'Hello, <strong>world</strong>',
                    key: Key('lineThrough=$lineThrough'),
                    textStyle: TextStyle(
                      decoration: lineThrough
                          ? TextDecoration.lineThrough
                          : TextDecoration.none,
                      fontSize: 15.0,
                      height: 1.4,
                    ),
                  ),
```

**Comment by daohoangson** [comment]
I thought about this a bit more and actually there is a better workaround for v0.4 (v0.5 handles this better and doesn't need workaround). Instead of using `HtmlWidget.textStyle`, you should put it inside a `DefaultTextStyle`, something like this:

```dart
      DefaultTextStyle(
        child: HtmlWidget('Hello, <strong>world</strong>'),
        style: DefaultTextStyle.of(context).style.copyWith(
              decoration: lineThrough
                  ? TextDecoration.lineThrough
                  : TextDecoration.none,
              fontSize: 15.0,
              height: 1.4,
            ),
      ),
```
